### ISSUE TITLE
✨ Ownership detection for ERC721 side

### ISSUE THREAD

**Issue by Vectorized** [Vectorized/dn404#42]
For marketplaces like Opensea

- Upon initialization handshake, the mirror will try to get the owner from the base contract. If there is an owner, the owner will be updated to that owner, and a `OwnershipTransferred(address indexed oldOwner, address indexed newOwner)` event will be emitted. Store the current owner in the deployer slot.

- Upon logging a batch NFT mint / burn, try to get the owner from the base contract. If the owner is different from that of the deployer slot, a `OwnershipTransferred(address indexed oldOwner, address indexed newOwner)` will be emitted, and the owner in the deployer slot will be updated.

Not sure if this will work, someone help me try.

**Comment by lambdalf-dev** [comment]
Why not just have an `owner()` function that returns the owner of the DN404 base contract?

**Comment by wwhchung** [comment]
Opensea does not rely on owner or OwnershipTransferred because it’s easily spoofable.

**Comment by lambdalf-dev** [comment]
Event easily spoofable for sure, owner is what the contract considers owner.

**Comment by lambdalf-dev** [comment]
Proposed solution here:
https://github.com/Vectorized/dn404/pull/43

**Comment by wwhchung** [comment]
> Event easily spoofable for sure, owner is what the contract considers owner.

Yup, but OpenSea typically doesn't use this to group things under an address's 'created tokens' because that can also be faked.  i.e. I can hardcode the owner() to return as you, and have these tokens as 'created by you' if OpenSea trusted the contract return values as the source of provenance.

**Comment by lambdalf-dev** [comment]
Well, if I'm the owner of the contract, it makes sense that the items were created by me...

**Comment by wwhchung** [comment]
> Well, if I'm the owner of the contract, it makes sense that the items were created by me...

Not necessarily. I could create a contract with a fake owner function and inject tokens into your collection. Similar exploit has been used in the past to make fake beeples which is why rarible and opensea moved away from relying on owner.

**Comment by cygaar** [comment]
Yeah @wwhchung is correct here - it's extremely easy to game `owner()` to make it look like someone else deployed your contract

**Comment by TechnicallyWeb3** [comment]
Why is the owner needed at all?

**Comment by lambdalf-dev** [comment]
> Why is the owner needed at all?

For marketplaces collection management

**Comment by sonicsmith** [comment]
Looks like a good solution would be to implement [ERC-7015](https://eips.ethereum.org/EIPS/eip-7015) since Opensea uses this for proof of ownership. (https://docs.opensea.io/docs/contract-level-metadata)
